### PR TITLE
APPS-497 CWL runner

### PR DESCRIPTION
**Pull request by mhrvol**
Added  CWL runner 
More info here: https://docs.google.com/document/d/1mNzL1XTR9ni3hlJM6JjAkhRwBvODrerk-pKhgI4iSXM/edit?ts=60523522

Normalizer script is not used - https://github.com/common-workflow-language/cwl-utils/pull/63, as it seems that only draft is currently ready (CWL files will not be transformed to the latest version)

Notes:
An input file does not exist (or there isn't read access to that file), cwl-runner will fail.
`--version` is not implemented correctly - Which version should be shown?
Currently `dxCompiler.jar` is used when running the job - which has to be in the same folder as the cwl-runner.py. How to specify different one?

### CODE REVIEW

**Comment by mr-c** [comment]
Will `cwl_runner/arg_parser.py` be installed as `dx-cwl-runner` (or similar)?

**Comment by mhrvol** [comment]
> Will `cwl_runner/arg_parser.py` be installed as `dx-cwl-runner` (or similar)?

Why arg_parser? There is `dx-cwl-runner.py`, which serves the purpose of entry point. I guess it makes sense to rename it to `dx-cwl-runner`

**Comment by mr-c** [comment]
> Why arg_parser? There is `dx-cwl-runner.py`,

Did `dx-cwl-runner.py` not get added to this PR? I didn't see it at https://github.com/dnanexus/dxCompiler/pull/92/files

**Comment by mr-c** [comment]
> Normalizer script is not used - [common-workflow-language/cwl-utils#63](https://github.com/common-workflow-language/cwl-utils/pull/63), as it seems that only draft is currently ready (CWL files will not be transformed to the latest version)

Can you tell me more about the issue with the normalizer script?

**Comment by mhrvol** [comment]
> > Why arg_parser? There is `dx-cwl-runner.py`,
> 
> Did `dx-cwl-runner.py` not get added to this PR? I didn't see it at https://github.com/dnanexus/dxCompiler/pull/92/files

https://github.com/dnanexus/dxCompiler/pull/92/files#diff-3c9692b28469a591c5abcefad460a4eaaef917aecb8332670e31fdb64b315392

It should be here

**Comment by mhrvol** [comment]
> > Normalizer script is not used - [common-workflow-language/cwl-utils#63](https://github.com/common-workflow-language/cwl-utils/pull/63), as it seems that only draft is currently ready (CWL files will not be transformed to the latest version)
> 
> Can you tell me more about the issue with the normalizer script?

I assume we want to include cwl utils as dependency - is there a release with cwl_normalizer? 

Do I misunderstand this? Is there a different, suitable way to use this cwl_normalizer without including the whole cwl-utils in dxCompiler?

File: cwl_runner/dx_cwl_runner/input_utils.py [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+
+def get_new_dx_input(input_path: str) -> str:
+    basename = os.path.basename(input_path)
+    new_input_filename = f"{os.path.splitext(basename)[0]}.dx.json"
+    return new_input_filename
+
+
+def get_paths_referenced(process_file: str, log: Log) -> Optional[dict]:
+    deps = json.loads(run_cmd(f"cwltool --print-deps {process_file}"))
+    deps.get("secondaryFiles")
```
Comment (mr-c) [inline cwl_runner/dx_cwl_runner/input_utils.py:146]:
```suggestion
    return deps.get("secondaryFiles")
```

**Comment by jdidion** [comment]
I've got this working now for most tests. The only tests that won't run now are 1) the ones that dxCompiler can't compile/run due to cwltool/cwljava bugs and 2) tests with `SecondaryFileSchema` defined in their input parameters. For the latter, we're waiting on @mr-c to implement https://github.com/common-workflow-language/cwltool/issues/1556.

**Review by jdidion — APPROVED** [review]

**Pull request merged**